CHANGELOG.md

```diff
@@ -1,5 +1,16 @@
 ## Unreleased
 
+## 0.10.3 (April 30, 2024)
+
+IMPROVEMENTS:
```

Comment:
nit: Could we maybe create a dependency update section rather than repeating them below? Dep updates don't feel like they belong in improvements. We do something similar VSO e.g: https://github.com/hashicorp/vault-secrets-operator/releases/tag/v0.6.0